DependencyInjection/Compiler/CheckForMailerPass.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the FOSUserBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\UserBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Flex\Recipe;
+
+/**
+ * Checks to see if the mailer service exists.
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+class CheckForMailerPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        // if the mailer isn't needed, then no error needed
+        if (!$container->has('fos_user.mailer')) {
+            return;
+        }
+
+        if ($container->findDefinition('fos_user.mailer')->hasTag('fos_user.requires_swift')) {
```

Comment:
This condition always throw a error even if i have installed fosub and swift bundle.
Mailers in `vendor/friendsofsymfony/user-bundle/Resources/config/mailer.xml` has tag `fos_user.requires_swift` defined in xml.
You should check if swift is installed and add tag in compiler or in this line check if swift is installed. I think first option is better.

Reply:
Nice catch! I forgot the check to see if the `mailer` service is present. It should be better now. The entire purpose of the tags is for me (here in the compiler pass) to be able to know if the mailer is one of the mailers that "requires swift". I did this because I don't want to throw the error if Swift is *not* installed, but the user has changed the mailer to their own custom mailer. I only want to throw the error if Swift is *not* installed AND they are using one of the built-in mailers that needs Swift. Giving those 2 services the tag was the best way I could think of to "mark" them.

But there was a bug - it should be fixed :).